umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/spatial.clj

```diff
@@ -81,17 +92,32 @@
     (into {} (for [[k ^String v] (partition 2 (str/split orbit-string #":? "))]
                [(keyword k) (Double/parseDouble v)]))))
 
+(defn- parse-vertical-domains
+  "Parse the vertical domain from the ISO XML document. Vertical domains are encoded in an ISO XML
+  document as a single string like this:
+  \"Type: Some type Value: Some value\""
+  [doc]
+  (when-let [vertical-domains (select doc vertical-string-xpath)]
+    (for [vertical-domain vertical-domains]
```

Comment:
I'm guessing it's a problem because you're re-defining vertical-domains in your let. You're iterating over vertical-domains in your for loop, but then defining a var with the same name in the let, which is probably causing some weirdness. Try changing the var name in your let and see if that fixes things. If not, let me know and I can help you with this.